diff --git a/index.d.ts b/index.d.ts
index 1a3d198217c3038904c363b9982140d5a20e0ce4..3e41c5a89c1b8b11519ac974248cdc8dad3cbd22 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -146,7 +146,7 @@ declare global {
         keys(): AsyncIterableIterator<string>;
         values(): AsyncIterableIterator<FileSystemDirectoryHandle | FileSystemFileHandle>;
         entries(): AsyncIterableIterator<[string, FileSystemDirectoryHandle | FileSystemFileHandle]>;
-        [Symbol.asyncIterator]: FileSystemDirectoryHandle["entries"];
+        [Symbol.asyncIterator](): FileSystemDirectoryHandle["entries"];
         /**
          * @deprecated Old property just for Chromium <=85. Use `kind` property in the new API.
          */
